### DISCUSSION TITLE
Avoiding boilerplate on protected api routes

### DISCUSSION THREAD

**Discussion by raphaelbadia** [TanStack/query#598] [General]
Hello,

Every app I work on includes some part that requires authentication, ofter a JWT bearer token stored in a context.

SWR provides a convenient way to define a default fetcher function :  https://github.com/vercel/swr/issues/325#issuecomment-628924614

```jsx
const state = useContext(AuthContext);
// ...
<SWRConfig
    value={{
        fetcher: async (url) => {
            const res = await fetch(url, {
                headers: {
                    'Content-Type': 'application/json',
                    Authorization: `Bearer ${state.tokens?.access}`,
                },
            });
            return res.json();
        },
    }}
>
    {children}
</SWRConfig>
```

How could I do something similar with react-query ? I'd like to avoid listening to my context in every component I want to use react-query. 🤔

Thanks !

**Comment by tannerlinsley** [comment]
You can use something like axios’s request interceptors or make a custom proxy hook for useQuery that closes over your setup
